### PR TITLE
Replaces OKTA with Strike auth

### PR DESCRIPTION
**Pull request by ata-peppered**
This PR does the following:

- Replace the okta auth with strike auth 
- Store private key behind biometry gate
- Remove signature from header
- Store public key in secure storage without biometry (under email same way private used to be stored)  
- Remove forgot password flow + remove all okta related screens
- Separate email and password in signin
- Implement hasPrivateKey for email method - does NOT require biometry
- On signin screen if user has key a private key it uses SignatureBased login otherwise PasswordBased login
- Make sure date format is correct `yyyy-MM-dd'T'HH:mm:ss.SSSxxx`
- Store email in AppStorage so it persists during app launch and user doesnt have to type it in every time 
- Make sure private key is used only once during encoding to avoid multiple biometry
- Remove all okta libraries and environment variables

**Migration path considered**
We were previously storying two items in the Keychain: 
- Private key for a given email
- Seed phrase for a given email

Both these values will now be stored under a biometry gate for the CURRENT biometry; meaning if biometry changes these values will be removed
Accessing these values should prompt for biometry 
The algorithm is as follows:
```
1. method takes in private key, root seed, and email
2. it stores private key for the email under biometry
3. it stores root seed for that email under biometry
4. it stored the public key for that private key for that email WITHOUT biometry
```
step 4 ensures a smooth experience for users joining the app

In order to improve the experience we need to store the public key alongside the private key;
This is done in the Keychain PER EMAIL. Same way we used to store Private Key before WITHOUT biometry
The is to used during authenticated state check where we we previously used to compare public keys with the backend to determine if user needs to register. 
The algorithm for this is as follows:
```
1. look inside secure storage for a public key for the given email, if it exists AND there exists a corresponding private key (without actually pulling it), just return it
2. if it doesnt exists look for the private (stored behind biometry) for the given email
3. if found grab the public key for it and store the public key WITHOUT biometry for this email
4. return the public key
```
The above creates migration path for existing users and adds a contingency should this public not exist

In the keychain a public is ONLY every returned IF a corresponding private key for that email also exists
this ensures we dont accidentally allow users into the app where they have a public key but no corresponding private key - just a safety net

image
image

### CODE REVIEW

File: Strike/Views/Auth/SignInView.swift [hunk 1]
```diff
@@ -6,79 +6,171 @@
...
-                            .foregroundColor(Color.black)
-                            .accentColor(Color.Strike.purple)
-                            .textFieldStyle(LightRoundedTextFieldStyle())
-                            //.colorScheme(.dark)
-
-                            Button("Forgot password?") {
-                                currentSheet = .recoverPassword
-                            }
-                            .foregroundColor(Color.Strike.purple)
+                        SecureField("", text: $password) {
```
Comment (bholzman) [inline Strike/Views/Auth/SignInView.swift:167]: we are also going to need to give the user the option to enter their recovery phrase if they don't know their password (but fine to do that in a subsequent change)

**Review by bholzman — APPROVED** [review]
Looks really great to me!

File: Strike/Networking/Auth/AuthProvider.swift [hunk 2]
```diff
@@ -11,7 +11,7 @@ protocol AuthProvider: AnyObject {
     var isAuthenticated: Bool { get }
     var isExpired: Bool { get }
     var bearerToken: String? { get }
-    var email: String? { get }
+    //var email: String? { get }
```
Comment (bflood-strike) [inline Strike/Networking/Auth/AuthProvider.swift:14]: you want to remove this

File: Strike/Networking/Moya+StrikeClient.swift [hunk 3]
```diff
@@ -11,7 +11,7 @@ import Moya
 extension DateFormatter {
     static let iso8601Full: DateFormatter = {
         let formatter = DateFormatter()
-        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ"
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSxxx"
```
Comment (bflood-strike) [inline Strike/Networking/Moya+StrikeClient.swift:14]: is the `xxx` intentional here

**Review by bflood-strike — APPROVED** [review]
looks great

File: Strike/Networking/Moya+StrikeClient.swift (hunk 3 above)
Comment (ata-peppered) [reply Strike/Networking/Moya+StrikeClient.swift:14]: Yes; this is the format that Brooklyn uses - the difference between ZZZZZ and xxx is that ZZZZZ will put a Z at the tail end when the timezone is GMT but xxx will put a +00:00.

File: Strike/Networking/Auth/AuthProvider.swift (hunk 2 above)
Comment (ata-peppered) [reply Strike/Networking/Auth/AuthProvider.swift:14]: yup

**Pull request merged**